### PR TITLE
Sprint 1 Oier & Yang

### PR DESCRIPTION
**Pull request by yang-uu**
- migrated from `doctest` to `unittest` for `lzw.py` and `rijndael.py`
- fixed unsolved reference in `dumppdf.py`

P.S. the sample pdf file was from [the Gutenberg Project](https://www.gutenberg.org/ebooks/19238), so hopefully there were not copyright issues. :)

### CODE REVIEW

File: tests/test_lzw.py
Comment (PrimeNumb) [inline tests/test_lzw.py]: Flake8 output:
```
.\test_lzw.py:11:80: E501 line too long (87 > 79 characters)
```

otherwise lgtm

File: tests/test_rijnadel.py
Comment (PrimeNumb) [inline tests/test_rijnadel.py:22]: 👍

File: tools/dumppdf.py
Comment (PrimeNumb) [inline tools/dumppdf.py:170]: Since checks were added relatively late, we ran these manually through flake8. This was the output for this particular file:
```
.\tools\dumppdf.py:18:1: F401 'pdfminer.utils.q' imported but unused
.\tools\dumppdf.py:22:1: E302 expected 2 blank lines, found 0
.\tools\dumppdf.py:40:15: E231 missing whitespace after ','
.\tools\dumppdf.py:71:80: E501 line too long (84 > 79 characters)
.\tools\dumppdf.py:94:1: E302 expected 2 blank lines, found 1
.\tools\dumppdf.py:102:1: E302 expected 2 blank lines, found 1
.\tools\dumppdf.py:107:32: E701 multiple statements on one line (colon)
.\tools\dumppdf.py:111:31: E701 multiple statements on one line (colon)
.\tools\dumppdf.py:122:1: E302 expected 2 blank lines, found 1
.\tools\dumppdf.py:127:22: E201 whitespace after '('
.\tools\dumppdf.py:127:56: E231 missing whitespace after ','
.\tools\dumppdf.py:128:62: E202 whitespace before ')'
.\tools\dumppdf.py:129:9: E306 expected 1 blank line before a nested definition, found 0
.\tools\dumppdf.py:141:23: E231 missing whitespace after ','
.\tools\dumppdf.py:141:29: E231 missing whitespace after ','
.\tools\dumppdf.py:141:34: E231 missing whitespace after ','
.\tools\dumppdf.py:141:36: E231 missing whitespace after ','
.\tools\dumppdf.py:150:80: E501 line too long (84 > 79 characters)
.\tools\dumppdf.py:153:80: E501 line too long (81 > 79 characters)
.\tools\dumppdf.py:168:1: E305 expected 2 blank lines after class or function definition, found 1
.\tools\dumppdf.py:170:1: E302 expected 2 blank lines, found 0
.\tools\dumppdf.py:182:80: E501 line too long (82 > 79 characters)
.\tools\dumppdf.py:198:80: E501 line too long (81 > 79 characters)
.\tools\dumppdf.py:203:1: E302 expected 2 blank lines, found 1
.\tools\dumppdf.py:213:24: E231 missing whitespace after ','
.\tools\dumppdf.py:225:30: E231 missing whitespace after ','
.\tools\dumppdf.py:233:5: E306 expected 1 blank line before a nested definition, found 0
.\tools\dumppdf.py:234:80: E501 line too long (88 > 79 characters)
.\tools\dumppdf.py:235:16: E127 continuation line over-indented for visual indent
.\tools\dumppdf.py:241:16: E701 multiple statements on one line (colon)
.\tools\dumppdf.py:252:21: E701 multiple statements on one line (colon)
.\tools\dumppdf.py:253:23: E701 multiple statements on one line (colon)
.\tools\dumppdf.py:254:23: E701 multiple statements on one line (colon)
.\tools\dumppdf.py:255:23: E701 multiple statements on one line (colon)
.\tools\dumppdf.py:255:40: E201 whitespace after '('
.\tools\dumppdf.py:255:71: E202 whitespace before ')'
.\tools\dumppdf.py:256:23: E701 multiple statements on one line (colon)
.\tools\dumppdf.py:256:39: E201 whitespace after '('
.\tools\dumppdf.py:256:68: E202 whitespace before ')'
.\tools\dumppdf.py:257:23: E701 multiple statements on one line (colon)
.\tools\dumppdf.py:258:23: E701 multiple statements on one line (colon)
.\tools\dumppdf.py:259:23: E701 multiple statements on one line (colon)
.\tools\dumppdf.py:260:23: E701 multiple statements on one line (colon)
.\tools\dumppdf.py:261:23: E701 multiple statements on one line (colon)
.\tools\dumppdf.py:274:1: E305 expected 2 blank lines after class or function definition, found 1
.\tools\dumppdf.py:274:26: E701 multiple statements on one line (colon)
```
you should address these before we approve.

File: pdfminer/lzw.py
Comment (PrimeNumb) [inline pdfminer/lzw.py]: Flake8 output:
```
.\pdfminer\lzw.py:9:1: E266 too many leading '#' for block comment
.\pdfminer\lzw.py:31:80: E501 line too long (81 > 79 characters)
.\pdfminer\lzw.py:68:13: E741 ambiguous variable name 'l'
.\pdfminer\lzw.py:69:16: E741 ambiguous variable name 'l'
.\pdfminer\lzw.py:71:18: E741 ambiguous variable name 'l'
.\pdfminer\lzw.py:73:18: E741 ambiguous variable name 'l'
.\pdfminer\lzw.py:99:1: W391 blank line at end of file
```

Fix these!

File: pdfminer/rijndael.py
Comment (PrimeNumb) [inline pdfminer/rijndael.py:1102]: Flake8 output:

```
.\pdfminer\rijndael.py:711:5: E303 too many blank lines (2)
.\pdfminer\rijndael.py:719:5: E303 too many blank lines (2)
.\pdfminer\rijndael.py:746:23: E701 multiple statements on one line (colon)
.\pdfminer\rijndael.py:764:22: E701 multiple statements on one line (colon)
.\pdfminer\rijndael.py:784:22: E701 multiple statements on one line (colon)
.\pdfminer\rijndael.py:809:21: E703 statement ends with a semicolon
.\pdfminer\rijndael.py:810:22: E703 statement ends with a semicolon
.\pdfminer\rijndael.py:812:25: E703 statement ends with a semicolon
.\pdfminer\rijndael.py:815:25: E703 statement ends with a semicolon
.\pdfminer\rijndael.py:816:30: E703 statement ends with a semicolon
.\pdfminer\rijndael.py:818:25: E703 statement ends with a semicolon
.\pdfminer\rijndael.py:819:30: E703 statement ends with a semicolon
.\pdfminer\rijndael.py:823:80: E501 line too long (89 > 79 characters)
.\pdfminer\rijndael.py:891:18: E701 multiple statements on one line (colon)
.\pdfminer\rijndael.py:994:18: E701 multiple statements on one line (colon)
.\pdfminer\rijndael.py:1085:1: W391 blank line at end of file
```

Fix these!

**Review by PrimeNumb — CHANGES_REQUESTED** [review]

**Review by PrimeNumb — CHANGES_REQUESTED** [review]
Also you really should take a look at the license for the test pdf - if this is copyrighted then we can't include it.

**Comment by yang-uu** [comment]
> Also you really should take a look at the license for the test pdf - if this is copyrighted then we can't include it.

it's from project gutenburg, the copyrights of books there have expired, so we shouldn't have any problems. :)
see the pr description for the link

File: pdfminer/lzw.py
Comment (PrimeNumb) [inline pdfminer/lzw.py:108]: Flake8 still failing on:
```
.\pdfminer\lzw.py:99:1: W391 blank line at end of file
```

**Review by PrimeNumb — CHANGES_REQUESTED** [review]

**Review by PrimeNumb — APPROVED** [review]

**Pull request merged**
